CMakeLists.txt

```diff
@@ -994,7 +994,24 @@ foreach(ZLIB_INSTALL_LIBRARY ${ZLIB_INSTALL_LIBRARIES})
 endforeach()
 
 if(WIN32)
-    set_target_properties(${ZLIB_INSTALL_LIBRARIES} PROPERTIES OUTPUT_NAME zlib${SUFFIX})
+    # Shared library
+    if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+        set_target_properties(zlib PROPERTIES OUTPUT_NAME zlib${SUFFIX})
+    endif()
+    # Static library
+    if(NOT DEFINED BUILD_SHARED_LIBS)
```

Comment:
Can you combine these four `set_target_properties` calls using `${ZLIB_INSTALL_LIBRARIES}` as the target?

Reply:
I don't think so...  `${ZLIB_INSTALL_LIBRARIES}` can also contain the shared library.

I tried combining matching ones using `zlibstatic` as target name but it did throw error that aliases can't be used with `set_target_properties`.

Reply:
What about using? `z$<$<C_COMPILER_ID:MSVC>:libstatic>${SUFFIX}`

Reply:
That's pretty much unreadable to average developer...